File: src/my_project/easy_problems/from201to250/reformat_string.py
from typing import List, overload 

class Solution:
    def reformat(self, s: str) -> str:
        
        def is_alpha(x):
            is_uppercase = ord(x) >= 65 and ord(x) <= 90
            is_lowercase = ord(x) >=97 and ord(x) <= 122
            return is_uppercase or is_lowercase
        def is_numeric(x):
            is_numeric = ord(x)>= 48 and ord(x) <= 57
            return is_numeric
        
        s1 = ''.join([x for x in s if is_alpha(x)])
        s2 = ''.join([x for x in s if is_numeric(x)])
        print(s1)
        print
        if abs(len(s1) - len(s2))> 1: 
            return ''
        
        
        if len(s1) < len(s2):
            s1, s2 = s2, s1
            
        answer = ''
        for i in range(len(s2)):
            answer += s1[i] + s2[i]
            
        if len(s1)> len(s2):
            answer += s1[-1]
        

        
        return answer 
    
solution = Solution()

print(solution.reformat(s = "a0b1c2"))

File: src/my_project_typescript/easy_problems_typescript/from101to150/reformatString.ts
function reformat(s: string): string {

    function isAlpha(x: string):boolean{
        let isUppercase: boolean = x.charCodeAt(0) >= 65 && x.charCodeAt(0) <= 90
        let isLowercase: boolean = x.charCodeAt(0) >= 97 && x.charCodeAt(0) <= 122
        return isUppercase || isLowercase
    }

    function isNumeric(x: string):boolean{
        let isNumeric: boolean = x.charCodeAt(0) >= 48 && x.charCodeAt(0) <= 57
        
        return isNumeric
    }

    let s1: string = ''
    let s2: string = ''
    for (let i = 0 ; i < s.length; i++){
        if (isNumeric(s[i])){
            s2+= s[i]
        }
        else{
            s1+=s[i]
        }
    }

    if (Math.abs(s1.length - s2.length) > 1){
        return ''
    }

    let temp1: string = ''
    let temp2: string = ''

    if (s1.length < s2.length){
        temp1 = s1 
        temp2 = s2 
        s1 = temp2
        s2 = temp1
    }

    let answer: string = ''
    for (let i = 0; i < s2.length; i++){
        answer +=s1[i] + s2[i]
    }

    if (s1.length > s2.length){
        answer += s1[s1.length-1]
    }
    

    return answer 
};


console.log(reformat("a0b1c2"))